src/jit/codegencommon.cpp

```diff
@@ -6209,40 +6211,51 @@ void CodeGen::genAllocLclFrame(unsigned frameSize, regNumber initReg, bool* pIni
         // The encoding differs based on the architecture and what register is
         // used (namely, using RAX has a smaller encoding).
         //
-        // loop:
         // For x86
-        //      test [esp + eax], eax       3
-        //      sub eax, 0x1000             5
-        //      cmp EAX, -frameSize         5
+        //      lea eax, [esp - frameSize]
+        // loop:
+        //      lea esp, [esp - pageSize]   7
+        //      test [esp], esp             3
```

Comment:
I think using rsp as a mask for test instruction has a good chance of performing poorly. I would be better to use rax as the mask, I think.

Reply:
Fixed